test/unit/model/authorization/TestTakerAuthorizationServiceTest.php

```diff
@@ -0,0 +1,300 @@
+                false
+            ],
+            'ActiveAndSecure' => [
+                'feature,security',
+                'http://www.tao.lu/Ontologies/TAODelivery.rdf#DeliveryExecutionStatusActive',
+                false
+            ],
+            'notActiveAndSecure' => [
+                'feature,security',
+                'http://www.tao.lu/Ontologies/TAODelivery.rdf#DeliveryExecutionStatusAuthorized',
+                false
+            ],
+            'notActiveAndSecure2' => ['security', 'state', false]
+        ];
+    }
+
+    /**
+     * @dataProvider getInvalidStates
+     * @param string $state
+     * @throws Exception
+     */
+    public function testVerifyResumeAuthorizationWithInvalidState($state)
+    {
+        $deliveryExecutionMock = $this->getDeliveryExecutionMock($state);
+        $this->service->setServiceLocator($this->getServiceLocatorMock([Ontology::SERVICE_ID => $this->ontologyMock]));
+
+        $this->expectException(UnAuthorizedException::class);
+        $this->service->verifyResumeAuthorization($deliveryExecutionMock, $this->getMock(User::class));
+
+    }
+
+    /**
+     * @return array
+     */
+    public function getInvalidStates()
+    {
+        return [
+            ['http://www.tao.lu/Ontologies/TAODelivery.rdf#DeliveryExecutionStatusFinished'],
+            ['http://www.tao.lu/Ontologies/TAODelivery.rdf#DeliveryExecutionStatusCanceled'],
+            ['http://www.tao.lu/Ontologies/TAODelivery.rdf#DeliveryExecutionStatusTerminated']
+        ];
+    }
+
+    /**
+     * @dataProvider isProctoredDataProvider
+     * @param core_kernel_classes_Property|null $propertyValue
+     * @param bool $proctorByDefault
+     * @param bool $expected
+     */
+    public function testIsProctored($propertyValue, $proctorByDefault, $expected)
+    {
+        $deliverySyncServiceMock = $this->getMock(DeliverySyncService::class);
+        $delivery = $this->getDeliveryMock();
+
+        $property = new core_kernel_classes_Property(
+            'http://www.tao.lu/Ontologies/TAODelivery.rdf#ProctorAccessible'
+        );
+
+        $this->ontologyMock->expects($this->once())->method('getResource')->with('deliveryUri');
+
+        $this->ontologyMock->expects($this->once())
+            ->method('getProperty')
+            ->with('http://www.tao.lu/Ontologies/TAODelivery.rdf#ProctorAccessible')
+            ->willReturn($property);
+
+        $delivery->expects($this->once())
+            ->method('getOnePropertyValue')
+            ->with($property)
+            ->willReturn($propertyValue);
+
+        $deliverySyncServiceMock->method('isProctoredByDefault')->willReturn($proctorByDefault);
+
+        $this->service->setServiceLocator(
+            $this->getServiceLocatorMock(
+                [
+                    Ontology::SERVICE_ID => $this->ontologyMock,
+                    DeliverySyncService::SERVICE_ID => $deliverySyncServiceMock
+                ]
+            )
+        );
+
+        if ($expected) {
```

Comment:
The same, it's better to check `$this->assertEquals($expected, $result);` instead of using `if`

Reply:
fixed